src/NUnitEngine/nunit.engine.tests/Services/TestRunnerFactoryTests/Results/Net20SingleAssemblyExpectedRunnerResults.cs

```diff
@@ -0,0 +1,104 @@
+            }
+            if (processModel == ProcessModel.Default && domainUsage == DomainUsage.Multiple)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.Default && domainUsage == DomainUsage.Single)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.InProcess && domainUsage == DomainUsage.Default)
+            {
+                return new RunnerResult { TestRunner = typeof(TestDomainRunner) };
+            }
+            if (processModel == ProcessModel.InProcess && domainUsage == DomainUsage.None)
+            {
+                return new RunnerResult { TestRunner = typeof(LocalTestRunner) };
+            }
+            if (processModel == ProcessModel.InProcess && domainUsage == DomainUsage.Multiple)
+            {
+                return new RunnerResult { TestRunner = typeof(MultipleTestDomainRunner) };
+            }
+            if (processModel == ProcessModel.InProcess && domainUsage == DomainUsage.Single)
+            {
+                return new RunnerResult { TestRunner = typeof(TestDomainRunner) };
+            }
+            if (processModel == ProcessModel.Separate && domainUsage == DomainUsage.Default)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.Separate && domainUsage == DomainUsage.None)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.Separate && domainUsage == DomainUsage.Multiple)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.Separate && domainUsage == DomainUsage.Single)
+            {
+                return new RunnerResult { TestRunner = typeof(ProcessRunner) };
+            }
+            if (processModel == ProcessModel.Multiple && domainUsage == DomainUsage.Default)
+            {
+                return new RunnerResult { TestRunner = typeof(MultipleTestProcessRunner) };
+            }
+            if (processModel == ProcessModel.Multiple && domainUsage == DomainUsage.None)
+            {
+                return new RunnerResult { TestRunner = typeof(MultipleTestProcessRunner) };
+            }
+            if (processModel == ProcessModel.Multiple && domainUsage == DomainUsage.Multiple)
+            {
+                return new RunnerResult { TestRunner = typeof(MultipleTestProcessRunner) };
+            }
+            if (processModel == ProcessModel.Multiple && domainUsage == DomainUsage.Single)
+            {
+                return new RunnerResult { TestRunner = typeof(MultipleTestProcessRunner) };
+            }
```

Comment:
Would nested `switch` statements be clearer? Also below.

Reply:
Yeah, I considered both. My thinking was that if we ever wanted to add a third option to this (e.g. framework), then this would be clearer than trebly-nested switch statements. I think that's probably overthinking things though...I think I'll rewrite it. Thanks! 😄